python-package/xgboost/compat.py

```diff
@@ -24,6 +26,84 @@ def py_str(x):
         """convert c string back to python string"""
         return x
 
+########################################################################################
+# START NUMPY PATHLIB ATTRIBUTION
+########################################################################################
+# Pathlib compatability used in  Numpy: https://github.com/numpy/numpy/tree/v1.17.0
+# Attribution:
+# https://github.com/numpy/numpy/blob/v1.17.0/numpy/compat/py3k.py#L15
+# pathlib
+try:
+    from pathlib import Path, PurePath  # noqa
+except ImportError:
+    Path = PurePath = None
+
+# Attribution:
+#https://github.com/numpy/numpy/blob/v1.17.0/numpy/compat/py3k.py#L188-L247
+# backport abc.ABC
+if sys.version_info[:2] >= (3, 4):
```

Comment:
This test is not needed as we require >= 3.4.  Just use ABC.

Reply:
Good catch - will do.